pkg/csi/service/common/vsphereutil.go

```diff
@@ -171,6 +170,43 @@ func CreateBlockVolumeUtil(ctx context.Context, clusterFlavor cnstypes.CnsCluste
 				spec.ScParams.DatastoreURL)
 		}
 	}
+	if spec.StoragePolicyID != "" {
+		// Check storage policy compatibility.
+		var sharedDSMoRef []vim25types.ManagedObjectReference
+		for _, ds := range datastores {
+			sharedDSMoRef = append(sharedDSMoRef, ds.Reference())
+		}
+
+		compat, err := vc.PbmCheckCompatibility(ctx, sharedDSMoRef, spec.StoragePolicyID)
+		if err != nil {
+			return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
+				"failed to find datastore compatibility "+
+					"with storage policy ID %q. Error: %+v", spec.StoragePolicyID, err)
+		}
+
+		compatibleDsMoids := make(map[string]struct{})
+		for _, ds := range compat.CompatibleDatastores() {
+			compatibleDsMoids[ds.HubId] = struct{}{}
+		}
+
+		compatibleDsAccessible := false
+		for _, ds := range datastores {
+			if _, exists := compatibleDsMoids[ds.Reference().Value]; exists {
+				compatibleDsAccessible = true
+				break
+			}
+		}
+		if !compatibleDsAccessible {
+			return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
```

Comment:
Is this a CSIInternalFault or can it classified as nonCSI fault ? Can you explain the corrective action for the users when they observe this error

Reply:
used newly introduced fault "CSIInvalidStoragePolicyConfigurationFault" instead of CSIInternalFault